### DISCUSSION TITLE
SPI lock disabled & SPI BIOS region unlocked

### DISCUSSION THREAD

**Discussion by sjuk** [fwupd/fwupd#7986] [Q&A]
**Describe the question**
The following tests failed and I wonder if there is anything I can do to improve the situation.
I followed the links from (--show-all) and searched the internet, but I didn't found any information on how to solve the issue.
The latest BIOS version is already installed and I didn't found any setting in the BIOS to enable a lock of the SPI.

```shell
fwupdmgr security --show-all
[..]
✘ SPI lock:                      Disabled: https://fwupd.github.io/libfwupdplugin/hsi.html#org.fwupd.hsi.Spi.Ble
✘ SPI BIOS region:               Unlocked: https://fwupd.github.io/libfwupdplugin/hsi.html#org.fwupd.hsi.Spi.SmmBwp
[..]
```

**fwupd version information**

```shell
fwupdmgr --version
compile   com.hughsie.libxmlb           0.3.18
compile   com.hughsie.libjcat           0.2.0
compile   org.freedesktop.fwupd         1.9.24
runtime   org.freedesktop.fwupd-efi     1.4
compile   org.freedesktop.gusb          0.4.8
runtime   com.hughsie.libxmlb           0.3.x
runtime   com.hughsie.libjcat           0.2.0
runtime   org.freedesktop.gusb          0.4.8
runtime   org.kernel                    6.8.0-47-generic
runtime   org.freedesktop.fwupd         1.9.24
```

**Comment by sjuk** [comment]
FYI: I've asked the manufacturer of the motherboard for help.

**Reply by sjuk** [reply]
Here is the answer from the vendor: Intel 9th ​​Gen Core processors are **not** affected.
Reference: [Intel SPI Flash Flaw Lets Attackers Alter or Delete BIOS/UEFI Firmware](https://www.bleepingcomputer.com/news/security/intel-spi-flash-flaw-lets-attackers-alter-or-delete-bios-uefi-firmware/)
I have an Intel i3-9100 CPU which is Gen 9.

**Reply by hughsie** [reply]
It might not be affected by that CVE, but it's still unlocked... You might get more information running a tool like chipsec.

**Comment by sjuk** [comment]
Answer from the manufacturer of the motherboard (Gigabyte): "We regret to inform you that due to the age of the motherboard, our team is no longer able to develop or maintain new BIOS versions. We ask for your understanding."

**Comment by peterzabriskie** [comment]
I have the same problem. On a brand new MSI motherboard. Which after installation i now find updates only with windows 10/11. But as it is new and all the components are new why I am getting these red flags.

I am also trying to run down where this /etc/dconf file has gone missing. It seems to be a recurring problem in Ubuntu Studio since 20.04 which no one has pinned down. https://forum.ubuntuusers.de/topic/dconf-warnungen/
```
(fwupdmgr:3979): dconf-WARNING **: 19:23:35.070: unable to open file '/etc/dconf/db/site': Failed to open file “/etc/dconf/db/site”: open() failed: No such file or directory; expect degraded performance
Idle…: 0%
Host Security ID: HSI:0! (v1.9.27)

HSI-1
✔ BIOS firmware updates:         Enabled
✔ csme manufacturing mode:       Locked
✔ csme override:                 Locked
✔ csme v0:16.1.30.2361:          Valid
✔ Platform debugging:            Disabled
✔ SPI write:                     Disabled
✔ Supported CPU:                 Valid
✔ TPM empty PCRs:                Valid
✔ TPM v2.0:                      Found
✔ UEFI bootservice variables:    Locked
✔ UEFI platform key:             Valid
✔ UEFI secure boot:              Enabled
✘ SPI lock:                      Disabled
✘ SPI BIOS region:               Unlocked

HSI-2
✔ Intel BootGuard:               Enabled
✔ Intel BootGuard OTP fuse:      Valid
✔ IOMMU:                         Enabled
✔ Platform debugging:            Locked
✔ TPM PCR0 reconstruction:       Valid
✘ Intel BootGuard ACM protected: Invalid
✘ Intel BootGuard verified boot: Invalid

HSI-3
✔ CET Platform:                  Supported
✔ Pre-boot DMA protection:       Enabled
✘ Intel BootGuard error policy:  Invalid
✘ Suspend-to-idle:               Disabled
✘ Suspend-to-ram:                Enabled

HSI-4
✔ SMAP:                          Enabled
✘ Encrypted RAM:                 Disabled

Runtime Suffix -!
✔ fwupd plugins:                 Untainted
✔ CET OS Support:                Supported
✔ Linux kernel lockdown:         Enabled
✔ Linux kernel:                  Untainted
✘ Linux swap:                    Unencrypted

This system has a low HSI security level.
 » https://fwupd.github.io/hsi.html#low-security-level

This system has HSI runtime issues.
 » https://fwupd.github.io/hsi.html#hsi-runtime-suffix

Host Security Events
  2025-01-01 23:01:11:  ✔ CET OS Support changed: Not supported → Supported
```
```
Operating System: Ubuntu Studio 24.04
KDE Plasma Version: 5.27.11
KDE Frameworks Version: 5.115.0
Qt Version: 5.15.13
Kernel Version: 6.8.0-51-lowlatency (64-bit)
Graphics Platform: Wayland
Processors: 20 × Intel® Core™ i5-14600K
Memory: 31.1 GiB of RAM
Graphics Processor: Mesa Intel® Graphics
Manufacturer: Micro-Star International Co., Ltd.
Product Name: MS-7D98
System Version: 3.0
```

**Reply by hughsie** [reply]
> But as it is new and all the components are new why I am getting these red flags.

Some vendors don't really care about security I'm afraid -- even on new and expensive hardware.

**Reply by peterzabriskie** [reply]
@hughsie can you expand on 'mechanical' issue the vendor is avoiding which creates a failure like this? I am under warranty I could pull the board and return it and go to another vendor. Or are the odds I will run into this again?

**Reply by hughsie** [reply]
It's not mechanical, it's about system design. e.g. see https://fwupd.github.io/libfwupdplugin/hsi.html
